pkg/apiserver/registry/inventory/vpc_test.go

```diff
@@ -0,0 +1,233 @@
+	"antrea.io/nephe/pkg/controllers/inventory"
+)
+
+var _ = Describe("VPC", func() {
+	var vpcIndexer1 = cache.NewIndexer(
+		func(obj interface{}) (string, error) {
+			vpc := obj.(*runtimev1alpha1.Vpc)
+			return fmt.Sprintf("%v/%v-%v", vpc.Namespace, vpc.Labels[inventory.VpcLabelAccountName], vpc.Info.Id), nil
+		},
+		cache.Indexers{
+			inventory.VpcIndexerByAccountNameSpacedName: func(obj interface{}) ([]string, error) {
+				vpc := obj.(*runtimev1alpha1.Vpc)
+				return []string{vpc.Namespace + "/" + vpc.Labels[inventory.VpcLabelAccountName]}, nil
+			},
+			inventory.VpcIndexerByVpcNamespacedName: func(obj interface{}) ([]string, error) {
+				vpc := obj.(*runtimev1alpha1.Vpc)
+				return []string{vpc.Namespace + "/" + vpc.Name}, nil
+			},
+			inventory.VpcIndexerByNamespacedRegion: func(obj interface{}) ([]string, error) {
+				vpc := obj.(*runtimev1alpha1.Vpc)
+				return []string{vpc.Namespace + "/" + vpc.Info.Region}, nil
+			},
+			inventory.VpcIndexerByNamespace: func(obj interface{}) ([]string, error) {
+				vpc := obj.(*runtimev1alpha1.Vpc)
+				return []string{vpc.Namespace}, nil
+			},
+		})
+
+	var l logger.Logger
+	cacheTest1 := &runtimev1alpha1.Vpc{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: "default",
+			Name:      "targetId",
+		},
+		Info: runtimev1alpha1.VpcInfo{
+			Id:   "targetId",
+			Name: "targetName",
+			Tags: map[string]string{
+				"no.delete": "true",
+			},
+			Cidrs: []string{"192.168.1.1/24"},
+		},
+	}
+	cacheTest2 := &runtimev1alpha1.Vpc{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: "non-default",
+			Name:      "targetId-nondefault",
+		},
+		Info: runtimev1alpha1.VpcInfo{
+			Id:   "targetId-Non",
+			Name: "targetName",
+			Tags: map[string]string{
+				"no.delete": "true",
+			},
+			Cidrs: []string{"192.168.1.1/24"},
+		},
+	}
+	cacheTest3 := &runtimev1alpha1.Vpc{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: "default",
+			Name:      "targetId2",
+			Labels: map[string]string{
+				inventory.VpcLabelAccountName: "accountname",
+				inventory.VpcLabelRegion:      "region",
+			},
+		},
+		Info: runtimev1alpha1.VpcInfo{
+			Id:   "targetId2",
+			Name: "targetName",
+			Tags: map[string]string{
+				"no.delete": "true",
+			},
+			Region: "region",
+			Cidrs:  []string{"192.168.1.1/24"},
+		},
+	}
+
+	cachedVpcs := []*runtimev1alpha1.Vpc{
+		cacheTest1,
+		cacheTest2,
+		cacheTest3,
+	}
+
+	expectedVpcs := []*runtimev1alpha1.Vpc{
+		cacheTest1,
+		nil,
+		cacheTest3,
+	}
+
+	Describe("Test Get function of Rest", func() {
+		for i, cachedVpc := range cachedVpcs {
+			_ = vpcIndexer1.Update(cachedVpc)
+			rest := NewREST(vpcIndexer1, l)
+			actualVPC, err := rest.Get(request.NewDefaultContext(), cachedVpc.Name, &metav1.GetOptions{})
+			if actualVPC == nil {
+				Expect(err).To(Equal(errors.NewNotFound(runtimev1alpha1.Resource("vpc"), cachedVpc.Name)))
+				Expect(actualVPC).Should(BeNil())
+				continue
+			}
+			Expect(err).Should(BeNil())
```

Comment:
what happens when in your cachedVPcs cacheTest2 comes at the end. you will have error and unit will fail. Modify the test and explicitly check for actualVPC is nil for cachetest2 only